core/src/modules/ops/destination.ts

```diff
@@ -1005,7 +1012,31 @@ WHERE
     try {
       for (const consideredExport of exportsWithChanges) {
         if (!mostRecentExportIds.includes(consideredExport.id)) {
-          await cancelOldExport(consideredExport);
+          await cancelExport(
+            consideredExport,
+            `Replaced by more recent export`,
+            "info"
+          );
+          continue;
+        }
+
+        const mostRecentExport = await Export.findOne({
+          where: {
+            destinationId: destination.id,
+            recordId: consideredExport.recordId,
+            state: "complete",
+          },
+          order: [["completedAt", "desc"]],
+        });
```

Comment:
this method will run for every `(const consideredExport of exportsWithChanges)`, which could be a lot of SQL statements.  We'll want to load this data in one SQL statement before the for loop for every export we are considering.